code/modules/projectiles/guns/ballistic.dm

```diff
@@ -56,8 +56,72 @@
 	var/tac_reloads = TRUE //Snowflake mechanic no more.
 	var/can_be_sawn_off  = FALSE
 
+	// stores all types of feedback. Name of feedback + the amount of frames in the animation (must be more than 0) . Add to this list as more feedback is added. 
+	var/list/feedback_types = list(
```

Comment:
Why not use the [flick(state_name) proc](https://secure.byond.com/docs/ref/#/proc/flick) instead of doing a memory intensive weird list thing?

Reply:
By doing it this way I can store the name of the action + the number of frames in the animation in 1 list instead of making 2 separate lists or define it as 2 separate vars in every gun.
Let's say a gun has a fire animation that lasts 3 frames or a magazine animation that lasts 12 they can do "fire" = 3 , "mag_out" = 12   

This way it also accomplishes 3 actions because I use the name to pull up the correct icon states too

Reply:
You dont need to store the length of the animation at all.... the flick() proc allows you to play an animation as long as it lasts and reverts back to the normal icon once its over

Reply:
Not sure.
Probably didn't use flick because some of the animations don't revert to the original icons. R/n the guns are built to use overlays and update those overlays on update/icon. The original gun under icon_state doesn't change since its the base of the gun.

Otherwise unsure both solutions might work.

For flick, I'd imagine the slide moving back or locking wouldn't work since it would pop back instead of changing the overlay icon state.

Reply:
Also, how is 5 items on a list "memory intensive" you're being overdramatic

Reply:
Its 5 items on a list, per instance, which adds code to initialize 1 more list per instance, on BYOND's internal init proc

Reply:
flick() will always revert to the original icon state when the animation is over. Thats simply how flick works and is used on many objects in game like telepads, airlocks, etc

Reply:
Over the last few days, I tried it a few times and couldn't get it to work with flick() without wonky behavior. (slide disappearing, timing issues  etc)
Turning this in to flick seems to require 2 separate systems for the same feature since all of them will not work on flick().

I tried; I won't be doing that part. 
This PR is a "take it or leave it" type deal unless there are other fixes I need to perform beside the flick() part.